cpp/src/io/utilities/datasource.cpp

```diff
@@ -110,7 +110,17 @@ class kvikio_source : public datasource {
     CUDF_EXPECTS(supports_device_read(), "Device reads are not supported for this file.");
 
     auto const read_size = std::min(size, this->size() - offset);
-    return _kvikio_handle.pread(dst, read_size, offset);
+
+    if constexpr (std::is_same_v<HandleT, kvikio::FileHandle>) {
+      return _kvikio_handle.pread(dst,
+                                  read_size,
+                                  offset,
+                                  kvikio::defaults::task_size(),
+                                  kvikio::defaults::gds_threshold(),
+                                  false);
+    } else if constexpr (std::is_same_v<HandleT, kvikio::RemoteHandle>) {
```

Comment:
looks like CUDF_KVIKIO_REMOTE_IO is not defined in jni. I think this line can simply be
```suggestion
    } else {
```
to avoid this dependency on the remote IO type

Reply:
True. Addressed